### PR TITLE
Issue #3753: Don't use a customizable date format in the dblog.

### LINKED ISSUE #3753: [UX] Add timestamp to date displayed in DBlog
I would like to include the time, as well as the date, in the 'Date' column of the dblog. 

Screen Shot 2019-05-03 at 4 53 13 PM

When something goes wrong on your site, its helpful to be able to check the date and the time of the problem, and then the date and time in the logs. If there's no time in the log, that's not so nice. 

If you have adjusted your date formats so that 'short' no longer includes a time, then you end up in a pickle when reviewing the dblog.

I think this might be one place where it would be better not to use a configurable date format.

**Steps to reproduce:**

1) go to Date and time formats (admin/config/regional/date-time)
2) remove the time from short and save
3) visit your error log (http://backdrop.local/admin/reports/dblog)

Problem:

* no timestamps

Expectation

* system logs not affected by front-end facing date format configuration

---
PR: https://github.com/backdrop/backdrop/pull/2645

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/3753

### CODE REVIEW

**Comment by findlabnet** [comment]
Sorry for late reaction, but hours format should be fixed
from `$dblog_date_format = settings_get('dblog_date_format', 'Y-M-d h:i');` - 12-hour format of an hour with leading zeros
to `$dblog_date_format = settings_get('dblog_date_format', 'Y-M-d H:i');` - 24-hour format of an hour with leading zeros

**Comment by herbdool** [comment]
@findlabnet good catch. I've updated it.

**Comment by herbdool** [comment]
I've just been debugging something inrelated and now I think we also should include seconds. Sometimes we need to see that level of detail.

File: core/modules/dblog/tests/dblog.test [hunk 1]
```diff
@@ -46,6 +46,9 @@ class DBLogTestCase extends BackdropWebTestCase {
 
     $row_limit = 1000;
     $this->verifyRowLimit($row_limit);
+    // Remove seconds from date so this test is more likely to always pass.
+    $date_format = 'M d, Y -  j:ia';
```
Comment (quicksketch) [inline core/modules/dblog/tests/dblog.test]: Throughout this test, there is an extra space in the date format between the dash and j: `-  j`

Comment (jenlampton) [reply core/modules/dblog/tests/dblog.test]: fixed.

File: core/modules/dblog/dblog.module [hunk 2]
```diff
@@ -158,6 +158,29 @@ function dblog_form_system_logging_settings_alter(&$form, $form_state) {
...
+
+  $format = config_get('system.core', 'log_date_format');
+  $pattern = isset($form_state['values']['dblog_date_format']) ? $form_state['values']['dblog_date_format'] : $format;
+  $preview = !empty($pattern) ? t('Displayed as %date', array('%date' => format_date(REQUEST_TIME, 'custom', $pattern))) : '';
+
+  $form['dblog_date_format'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Date format for log messages'),
+    '#maxlength' => 100,
+    '#description' => t('A date format using PHP date and time codes. See the <a href="@url">PHP manual</a> for available options.', array('@url' => 'http://php.net/manual/function.date.php')),
```
Comment (klonos) [inline core/modules/dblog/dblog.module:170]: Less verbose:

```suggestion
    '#description' => t('A date format using <a href="@url">PHP date and time codes</a>.', array('@url' => 'http://php.net/manual/function.date.php')),
```

File: core/modules/dblog/dblog.install [hunk 3]
```diff
@@ -99,3 +99,11 @@ function dblog_update_1000() {
     db_create_table('watchdog', backdrop_get_schema_unprocessed('dblog', 'watchdog'));
   }
 }
+
+/**
+ * Set the format for dates in the database log.
+ */
+function dblog_update_1001() {
+  $short = system_get_date_formats('short');
```
Comment (quicksketch) [inline core/modules/dblog/dblog.install]:
```suggestion
  $short = system_date_format_load('short');
```

It results in the same thing, but getting a single format back from "get_date_format**s**" makes me cringe.

**Comment by quicksketch** [comment]
Merged into 1.x and 1.13.x.

**Pull request merged**